src/Controller/SentencesListsController.php

```diff
@@ -169,13 +170,20 @@ public function show($id = null, $translationsLang = null)
         $this->loadModel('Sentences');
         $this->loadModel('SentencesSentencesLists');
 
-        $query = $this->SentencesSentencesLists->find()
-            ->where(['sentences_list_id' => $id])
+        $query = $this->SentencesSentencesLists->find();
+        if ($lang!="und") {
+            if ($lang == 'unknown') {
+                $query->where(['lang IS' => null]);
+            } else {
+                $query->where(['lang' => $lang]);
```

Comment:
If I remember correctly, I think CakePHP will automatically check for parameter nullity and add the `IS` keyword for you. Have you tried passing `['lang' => $lang]` with $lang being null?

Reply:
The if else block is to catch the case where `lang` is `unknown` not `null`. So we can't directly make use of cakePHP's automatic null check during query generation. I could however use a ternary to convert `unknown` to `null` then use just one query. 

`$query->where(['lang IS' => $lang==='unknown'?null:$lang]);`